skybar/src/main/java/org/wtf/skybar/transform/SkybarTransformer.java

```diff
@@ -20,12 +24,26 @@ public SkybarTransformer(String prefix) {
             System.out.println("Instrumenting " + className);
             ClassReader reader = new ClassReader(bytes);
             ClassWriter writer = new ClassWriter(reader, 0);
-            reader.accept(new SkybarClassVisitor(writer), ClassReader.EXPAND_FRAMES);
+            ClassVisitor visitor = writer;
+            if(shouldTrace()) {
+                visitor = new TraceClassVisitor(writer, new PrintWriter(System.out));
+                System.out.println("Unchanged bytecode: ");
+                reader.accept(new TraceClassVisitor(new PrintWriter(System.out)), ClassReader.EXPAND_FRAMES);
+            }
+            try {
+                reader.accept(new SkybarClassVisitor(visitor), ClassReader.EXPAND_FRAMES);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
             return writer.toByteArray();
         }
         return bytes;
     }
 
+    private boolean shouldTrace() {
+        return false;
```

Comment:
This is most useful for development currently. But it will become useful for creating debug reports once we have users who are not Skybar developers. I would like to see config options for this, using include/exclude patterns in symmetry with source file includes.